### PR TITLE
feat: add C implementation of `stats/base/dists/arcsine/logpdf`

### LINKED ISSUE #3401: [RFC]: Add C implementation for `@stdlib/stats/base/dists/arcsine/logpdf`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/arcsine/logpdf`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/arcsine/logpdf) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify that C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/arcsine/logpdf" make install-node-addons
```

#### Run C Examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/benchmark/benchmark.native.js"
```

#### Run All Tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/arcsine/logpdf/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Vinit-Pandit**
Resolves #3401 

## Description

> What is the purpose of this pull request?

This pull request:

-    add c implementation of `stats/base/dists/arcsine/logpdf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3401

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this, @MeastroZI. Looks like you are missing changes to the package README.

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/include/stdlib/stats/base/dists/arcsine/logpdf.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+* Evaluates the logarithm of the probability density function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
+*/
+double stdlib_base_dists_arcsine_logpdf( const double x, const double a, const double b );
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif // !STDLIB_STATS_BASE_DISTS_ARCSINE_CDF_H
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/include/stdlib/stats/base/dists/arcsine/logpdf.h]:
```suggestion
#endif // !STDLIB_STATS_BASE_DISTS_ARCSINE_LOGPDF_H
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// MAIN //
+
+/**
+*  Evaluates the logarithm of the probability density function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
+*
+* @private
+* @param {number} x - input value
+* @param {number} a - minimum support
+* @param {number} b - maximum support
+* @returns {Probability} evaluated LOGPDF
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/lib/native.js]:
```suggestion
* @returns {number} evaluated logPDF
```

Beware of copy-paste errors. Check against the `main.js` file.

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/src/main.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/arcsine/logpdf.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/constants/float64/ninf.h"
+#include "stdlib/constants/float64/ln_pi.h"
+#include "stdlib/math/base/special/ln.h"
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/src/main.c]:
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/src/main.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+/**
+* Evaluates the logarithm of the probability density function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
+*
+* @param x    input value
+* @param a    minimum support
+* @param b    maximum support
+* @return     evaluated LOGPDF
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/src/main.c]:
```suggestion
* @return     evaluated logPDF
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/README.md [hunk 5]
```diff
@@ -147,6 +147,110 @@ for ( i = 0; i < 25; i++ ) {
...
+    double b;
+    double y;
+    int i;
+    
+    for ( i = 0; i < 25; i++ ) {
+        x = random_uniform( -10.0, 10.0 );
+        a = random_uniform( -20.0, 0.0 );
+        b = random_uniform( a, a+40.0 );
+        y = stdlib_base_dists_arcsine_logpdf( x, a, b );
+        printf( "x: %lf, a: %lf, b: %lf, F(x;a,b): %lf\n", x, a, b, y );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/README.md]:
```suggestion
        printf( "x: %lf, a: %lf, b: %lf, ln(f(x;a,b)): %lf\n", x, a, b, y );
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/src/main.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+/**
+* Evaluates the logarithm of the probability density function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
+*
+* @param x    input value
+* @param a    minimum support
+* @param b    maximum support
+* @return     evaluated logPDF
+*
+* @example
+* double y = stdlib_base_dists_arcsine_logpdf(  2.0, 0.0, 4.0 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/src/main.c]:
```suggestion
* double y = stdlib_base_dists_arcsine_logpdf( 2.0, 0.0, 4.0 );
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/lib/native.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+*  Evaluates the logarithm of the probability density function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/lib/native.js]:
```suggestion
* Evaluates the logarithm of the probability density function (PDF) for an arcsine distribution with minimum support `a` and maximum support `b` at a value `x`.
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/examples/c/example.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+	double b;
+	double y;
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		x = random_uniform( -10.0, 10.0 );
+		a = random_uniform( -20.0, 0.0 );
+		b = random_uniform( a, a+40.0 );
+		y = stdlib_base_dists_arcsine_logpdf( x, a, b );
+		printf( "x: %lf, a: %lf, b: %lf, F(x;a,b): %lf\n", x, a, b, y );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/logpdf/examples/c/example.c]:
```suggestion
		printf( "x: %lf, a: %lf, b: %lf, ln(f(x;a,b)): %lf\n", x, a, b, y );
```

**Review by Planeshifter — APPROVED** [review]
Thank you, @MeastroZI!

Fixed a few small things (in example printouts:`F` is used for the CDF, whereas `f` should be used for the PDF) but the PR looks good to me overall and should be ready to land! 🚀

**Pull request merged**